### PR TITLE
Add reproducibility

### PR DESCRIPTION
**Pull request by AndrzejKaj**
Add reproducibility by creating fields for loading json files, saving it in local storage and changing values of UI components on entry to each stage.

### CODE REVIEW

**Comment by AndrzejKaj** [comment]
[screening_settings_2023-11-26 (1).json](https://github.com/zuzg/drug-screening/files/13468149/screening_settings_2023-11-26.1.json)
[correlation_analysis_settings_2023-11-26 (1).json](https://github.com/zuzg/drug-screening/files/13468150/correlation_analysis_settings_2023-11-26.1.json)
[hit_validation_settings_2023-11-26.json](https://github.com/zuzg/drug-screening/files/13468151/hit_validation_settings_2023-11-26.json)

File: dashboard/pages/hit_validation/callbacks.py [hunk 1]
```diff
@@ -177,6 +179,51 @@ def on_file_upload(
...
+    top_lower_bound_value = top_lower_bound
+    top_upper_bound_value = top_upper_bound
+
+    if loaded_data != None:
+        concentration_lower_bound_value = loaded_data["concentration_lower_bound"]
+        concentration_upper_bound_value = loaded_data["concentration_upper_bound"]
+        top_lower_bound_value = loaded_data["top_lower_bound"]
+        top_upper_bound_value = loaded_data["top_upper_bound"]
+
+    print(loaded_data)
```
Comment (Tremirre) [inline dashboard/pages/hit_validation/callbacks.py:217]: remove redundant print

File: dashboard/pages/screening/callbacks.py [hunk 2]
```diff
@@ -215,11 +227,41 @@ def on_outlier_purge_stage_entry(
...
+    Callback for the stage 2 entry
+    Change value of checkbox if json was loaded
+
+    :param current_stage: current stage index of the process
+    :param value: z threshold, slider value
+    :param saved_data: dict with loaded data
+    :return: value for checkbox
+    """
+
+    print(value)
```
Comment (Tremirre) [inline dashboard/pages/screening/callbacks.py:245]: remove redundant print

File: dashboard/pages/screening/callbacks.py [hunk 3]
```diff
@@ -215,11 +227,41 @@ def on_outlier_purge_stage_entry(
...
+    if current_stage != 1:
+        return no_update
+
+    checkbox = value
+    if saved_data != None:
+        if saved_data["outliers_preview_stage"]["outliers_only_checklist"]:
```
Comment (Tremirre) [inline dashboard/pages/screening/callbacks.py:251]: combine these ifs

File: dashboard/pages/screening/callbacks.py [hunk 4]
```diff
@@ -215,11 +227,41 @@ def on_outlier_purge_stage_entry(
...
+) -> list[bool]:
+    """
+    Callback for the stage 2 entry
+    Change value of checkbox if json was loaded
+
+    :param current_stage: current stage index of the process
+    :param value: z threshold, slider value
+    :param saved_data: dict with loaded data
+    :return: value for checkbox
+    """
```
Comment (Tremirre) [inline dashboard/pages/screening/callbacks.py:243]: to be fair - imo this setting does not really need saving for later - it is **purely visual** and does not impact the processing of the data, which the settings systems has to facilitate

**Review by Tremirre — APPROVED** [review]
LGTM but haven't tested yet locally, gonna do it tomorrow
\+ added some comments

File: dashboard/pages/correlation/callbacks.py [hunk 5]
```diff
@@ -111,6 +113,27 @@ def on_both_files_uploaded(
...
+
+    :param content: base64 encoded file content
+    :param name: filename
+    :return: dict with loaded data
+    """
+    if not content:
+        return no_update
+    loaded_data = load_data_from_json(content, name)
+    color = "success"
+    text = "Settings uploaded successful"
```
Comment (azywot) [inline dashboard/pages/correlation/callbacks.py:128]: Settings uploaded ~successful~ -> successfully 
(the same with other cases)

File: dashboard/pages/correlation/callbacks.py [hunk 6]
```diff
@@ -111,6 +113,27 @@ def on_both_files_uploaded(
...
+    if loaded_data == None or not set(["concentration_value", "volume_value"]).issubset(
+        loaded_data.keys()
+    ):
+        color = "danger"
+        text = "Settings not uploaded"
```
Comment (azywot) [inline dashboard/pages/correlation/callbacks.py:133]: Maybe sth like `Invalid settings uploaded: the file should contain {settings keys} keys.`

**Review by azywot — APPROVED** [review]
LGTM 👍
added smol comments

**Pull request merged**